### PR TITLE
Feedback for fire lasting time.

### LINKED ISSUE #24977: Better presentation of information about fireplaces.
**Is your feature request related to a problem? Please describe.**  
Sometimes the fire gets out before the cooking is finished. Then the process stops, because lack of light. How does one know how long a 2-by-4, heavy stick, wood splinter, etc. burns?

Also in the description of the wood stove it´s says the wood stove is more efficient than an open fire. -The stone fire place is semy open, right?
So how much more efficient is the wood stove?

In the stone fireplace it says the fire is relaxing.
How much relaxation or /Joy /does the stone firepace give? 

How much is the efficiency difference there between the different woodfed fireplaces? Bracier, digged pit, wood stove, stone fireplace, etc.

There are some things which are not clear in the game.

**Describe the solution you'd like**  
All wood fed fireplaces could use a counter for how much units of "cooking" they have left in them. Like the survivor mess kit or gasoline cooker. Or would that be bad for the game?
As gasoline has very defined properties as a product, a counter on the cooker makes sense very much for the game.
Wood could be different piece by piece but I think it is not needed for the gameplay to give every piece of wood an RNG based fuel value.

So the cooking places could get different efficiency falues for using the energy value of the fuel which was put in.

Efficiency:
Wood stove > stone fireplace > fire pit > bracier ....
There are others like the charcoal cooker but I havent used everything in the game yet.

So butchering tools have a value for how good they are for that task, like butchering 22. Saws and other tools have their value in numbers also. How thinkable is it to introduce such a property for the fire places? Like "Fuel efficiency: 0,5"

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
`SUMMARY: Interface "Added feedback for contained fire's expected time left, before it goes out."`

#### Purpose of change
There was no feedback for expected time when fire will go out. This caused problems with fuel management, as you were not able to determine how much consumed fuel is "stored" in fire at any given time, and how it is translated to fire longevity. Only option was to have a guess by items put into the fire, but when they were consumed nothing else was relevant to do the measurement.

#### Describe the solution
There is an internal discrete timer that determines fire's life (and half-life when fire is dying out). I have translated that timer into a feedback that player can acquire by examining fire. Also:
- only contained fires (wood stoves, fireplaces, ovens etc.) ~~and fires in pits~~ give back information, so you cannot measure wild fires;
- survival skill plays a role in correct assessment, the better it is, the more precise the estimated time before fire goes out, with cap at level 5 (most accurate info);
- different info for time left < 30 minutes, a its actually form of half-life (max life) with random chance of dying out sooner.

#### Describe alternatives you've considered
I've considered a feature of calculating not only consumed fuel, but all other fuel in the fire in form of items, but that would require rewriting a big portion of old code in field.cpp so I did not go that way. It's is still possible to "manually" assess fuel value of some items, just by experimenting with this new feature.

#### Additional context
Resolves #24977

### CODE REVIEW

**Comment by EirenexTheDragon** [comment]
with low skills it could perhaps be just a guess of describing how the fire is looking,aswell of rough time. 
for example
"This fire looks like its about to die" for like flames that last a few minutes. 
that just got into my mind when looking this. good luck by the way

**Comment by nexusmrsep** [comment]
@ZhilkinSerg I never even considered it. If you think it's relevant I'd be glad to add it. But wouldn't it be strange to do such assesment from the distance?

**Comment by ZhilkinSerg** [comment]
We already view item list from the distance.

**Comment by nexusmrsep** [comment]
Ok, done. When conditions are met same text will be printed in mouse view.

**Comment by ZhilkinSerg** [comment]
There is too much variation in fire burning time estimate (from `about 1 minute` to `more than 5 hours`) for survivor with survival level 10.

**Comment by nexusmrsep** [comment]
> There is too much variation in fire burning time estimate (from about 1 minute to more than 5 hours) for survivor with survival level 10.

That is no variation, that is a bug. And I pushed a fix for it. Now the skill cap at which you get precise info is survival = 5. At 0-1 you get simplified info, at 2-4 estimation varies and get better on higher levels, most precise at lvl 5.

**Also testing showed some strange behavior (bug) not originating from this PR:**

1. Fire in a fire container (ex. brazier) will quite quickly convert all items into fuel, so examining fire gives you information about how long will it burn with values from minutes to infinity (apparently there is no fuel limit)
image

2. Fire in a shallow pit / pit will stay at top of its half-life value (around `30_minutes`) so examining fire fill gives you **static** information, but will convert items to fuel slowly - only when it's countdown falls to allow to consume some more (so one can measure it longevity by items placed inside), unless you add enough fuel for the fire to grow in size - and either spread, or burn it's fuel faster then returned `time_duration`. Whatever the outcome, this type of fire produces unexpected and immeasurable behavior so I have to drop support for fire in pits and shallow pits. Perhaps I may return to it if I get an angle on how to handle all variables.

**Comment by kevingranade** [comment]
> Whatever the outcome, this type of fire produces unexpected and immeasurable behavior so I have to drop support for fire in pits and shallow pits. Perhaps I may return to it if I get an angle on how to handle all variables.

That's fine, we only need estimates for constructed fires, not haphazard piles.

File: src/game.cpp [hunk 1]
```diff
@@ -7505,6 +7506,56 @@ void game::examine()
...
+                fire_age = fire_age * -1 + 30_minutes;
+                if( mod >= 4 ) { // = survival level 0-1
+                    if( fire_age <= 1_hours ) {
+                        ss << string_format( _( "It's quite decent and looks like it'll burn for a bit without extra fuel." ) );
+                        return ss.str();
+                    } else if( fire_age <= 3_hours ) {
+                        ss << string_format( _( "It's looks solid, and will burn for a few hours without extra fuel." ) );
+                        return ss.str();
+                    } else {
+                        ss << string_format( _( "It's very well supplied and even without extra fuel might burn for at least s part of a day." ) );
```
Comment (Night-Pryanik) [inline src/game.cpp]: `at least s part of a day.` What this means?

**Comment by nexusmrsep** [comment]
Typo. "A" part of a day.

pt., 7 wrz 2018, 17:08 użytkownik Anton Burmistrov <notifications@github.com>
napisał:

**Pull request merged**